planner/aamwriter/src/main/java/eu/seaclouds/platform/planner/aamwriter/modeldesigner/DGraph.java

```diff
@@ -122,13 +127,27 @@ public DNode getNode(String nodeName) {
         return result;
     }
 
+    private DNode calculateSourceNode() {
+        DNode result = DNode.NOT_FOUND;
+        
+        for (DNode node : nodes) {
+            List<DLink> links = getLinksTo(node);
+            if (links.size() == 0) {
+                result = node;
+                break;
+            }
+        }
+        return result;
+    }
+    

```

Comment:
@jacopogiallo : This is related with how the optimizer works. The optimizer assumes one single frontend (the one annotated with QoSAppRequirements). Chris is using such a graph, so I am adding the feature to select which node is the frontend from the UI (this PR is just the backend step). 

In summary, calculateSourceNode() is a fallback method, to avoid the aam generation fails. This worth a comment, indeed.